lib/bap_llvm/llvm_macho_loader.hpp

```diff
@@ -295,24 +311,40 @@ void iterate_macho_commands(const macho &obj, ogre_doc &s) {
         macho_command(obj, cmd, s);
 }
 
-void iterate_dyn_relocations(const macho &obj, std::size_t off, std::size_t num, ogre_doc &s) {
-    const char *ptr = prim::get_raw_data(obj);
-    std::size_t next = off;
+MachO::relocation_info get_rel(const macho &obj, uint32_t off) {
+    const char *ptr = prim::get_raw_data(obj) + off;
+    MachO::relocation_info rel;
+    if (obj.isLittleEndian() != sys::IsLittleEndianHost) {
+        MachO::any_relocation_info any;
+        memcpy(&any, ptr, sizeof(MachO::any_relocation_info));
+        MachO::swapStruct(any);
+        memcpy(&rel, &any, sizeof(MachO::relocation_info));
+    } else {
+        memcpy(&rel, ptr, sizeof(MachO::relocation_info));
+    }
+    return rel;
+}
+
+// Note, that is r_extern is set to 1, than r_symbolnum contains an index in symbtab, and
+// section index otherwise
+void iterate_dyn_relocations(const macho &obj, uint32_t off, uint32_t num, ogre_doc &s) {
+    uint32_t next = off;
     for (std::size_t i = 0; i < num; ++i) {
-        const MachO::relocation_info *rel = reinterpret_cast<const MachO::relocation_info *>(ptr + next);
-        if (rel->r_address & MachO::R_SCATTERED) {
+        MachO::relocation_info rel = get_rel(obj, next);
+        if (rel.r_address & MachO::R_SCATTERED) {
```

Comment:
`rel.r_address & MachO::R_SCATTERED` doesn't imply that the relocation `rel` is scattered, it also depends on the architecture, e.g., for x86_64 and arm64 the relocation would be an immediate value, i.e., 64 bit, so this iteration will go off track in case of 64 bit system. If needed I can give the same testing binary for the x86_64 arch.

There is a `isRelocationScattered` member-function of the `MachOObjectFile` class that can tell you for sure if it scattered or now. Is there any _good_ reason why you are not using this function? Also there are plenty of relocation related member functions that apparently takes the endianness into account. Why aren't you using them?